lib/DiscoveryStreamFeed.jsm

```diff
@@ -163,25 +163,84 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     }
   }
 
+  /**
+   * buildFeedPromise - Creates a promise for a single unique feed
+   *                    and populates newFeeds with the result.
```

Comment:
Maybe s/populates newsFeeds with the result/pushes the result onto the newsFeeds array/ for clarity...

Reply:
That's better than what's currently there, as I was writing it a tweaked it even more, how does this look?

```
   * buildFeedPromise - Pushes a promise to newsFeedsPromises and
   *                    populates newFeeds with the promise result.
```

Reply:
Reversed the order too:
```
   * buildFeedPromise - Populates newFeeds with the promise result and
   *                    pushes a promise to newsFeedsPromises.
```